File: Homework/JavaScript-Fundamentals/11. Strings/parse-url.js
function solve(args) {
    let str = args[0];

    let protocol = str.substring(0, str.indexOf('://'));
    let server = str.substring(str.indexOf('://') + 3, str.indexOf('/', str.indexOf('://') + 3));
    let resource = str.substring(str.indexOf(server) + server.length);

    console.log('protocol: ' + protocol);
    console.log('server: ' + server);
    console.log('resource: ' + resource);
}

File: Homework/CSharp-Part-2/06. Strings-and-Text-Processing/ParseURL.cs
using System;
using System.Text;

class ParseURL
{
    static void Main()
    {
        string input = Console.ReadLine();

        var url = new Uri(input);
        string protocol = url.Scheme;
        string server = url.Host;
        string resource = url.PathAndQuery;

        var formatted = "[protocol] = {0}\n[server] = {1}\n[resource] = {2}";
        Console.WriteLine(formatted, protocol, server, resource);
    }
}